request.go

```diff
@@ -41,40 +46,52 @@ type Request struct {
 
 // NewRequest returns a new Request, given an address, command, and list of
 // arguments.
-func NewRequest(addr string, cmd string, args Args) *Request {
+func NewRequest(addr string, cmds ...Command) *Request {
 	return &Request{
 		Addr: addr,
-		Cmd:  cmd,
-		Args: args,
+		Cmds: cmds,
 	}
 }
 
 // ParseArgs parses the list of arguments from the request into the destination
 // pointers, returning an error if something went wrong.
 func (req *Request) ParseArgs(dsts ...interface{}) error {
-	return ParseArgs(req.Args, dsts...)
+	for _, cmd := range req.Cmds {
+		err := ParseArgs(cmd.Args, dsts...)
+		if err != nil {
```

Comment:
Let's inline the error check here `if err := ParseArgs(cmd.Args, dsts...); err != nil {`